### PR TITLE
Add workaround for Microsoft::InfraManagers

### PR DESCRIPTION
**Pull request by MattLombana**
Workaround for bug: https://bugzilla.redhat.com/show_bug.cgi?id=1351253
    run rails console command

Change run_rails_console so that 'command' is interpreted literally

### CODE REVIEW

**Comment by MattLombana** [comment]
If run_rails_console( ) needs to be run with double quotes surrounding 'command' then let me know, so I can create method run_rails_command_literal( )

File: cfme-performance/utils/providers.py [hunk 1]
```diff
@@ -286,6 +287,20 @@ def add_provider(provider):
 
     logger.debug('Added Provider: {}, Response: {}'.format(provider['name'], response))
 
+    # Workaround for Bug: https://bugzilla.redhat.com/show_bug.cgi?id=1351253
+    if provider['type'] == "ManageIQ::Providers::Microsoft::InfraManager":
+        print "In if statement!!!"
+        time.sleep(10)
```
Comment (akrzos) [inline cfme-performance/utils/providers.py]: Is there a race condition you ran into without this sleep statement?

File: cfme-performance/utils/providers.py [hunk 2]
```diff
@@ -286,6 +287,20 @@ def add_provider(provider):
...
+        logger.debug('Running rails command for Microsoft InfraManager Workaround')
+        ssh_client = SSHClient()
+        command = ('m = ExtManagementSystem.find_by_name "' + provider['name'] +
```
Comment (akrzos) [inline cfme-performance/utils/providers.py]: My preference is on using the .format to put things into strings.  An example you might find useful for this case would be here:

https://github.com/akrzos/cfme-performance/blob/master/cfme-performance/utils/appliance.py#L279-L281

Another good example is here:

https://github.com/akrzos/cfme-performance/blob/master/cfme-performance/utils/appliance.py#L294-L300

**Comment by akrzos** [comment]
> If run_rails_console( ) needs to be run with double quotes surrounding 'command' then let me know, so I can create method run_rails_command_literal( )

I think we should keep it as double quotes to avoid breaking any other calls to that method.  I provided two examples of ruby code pushed into the rails console that have variables using single quotes.  Additionally using .format will help make the code more clear and consistent with our other string concatenation.

File: cfme-performance/utils/providers.py (hunk 1 above)
Comment (MattLombana) [inline cfme-performance/utils/providers.py]: Occasionally, the provider would not be fully processed by the API and added to the vmdb by the time it reached this step of execution

File: cfme-performance/utils/providers.py [hunk 3]
```diff
@@ -286,6 +287,20 @@ def add_provider(provider):
...
+        logger.debug('Running rails command for Microsoft InfraManager Workaround')
+        ssh_client = SSHClient()
+        command = ('m = ExtManagementSystem.find_by_name "{}"; attributes = {{:security_protocol =>'
+        ' "{}"}}; m.update_authentication(:default => {{:userid => "{}", :password => "{}"}}); '
+        'm.update_attributes(attributes); m.save; m.authentication_check;'
+        .format(provider['name'], provider['security_protocol'],
+            provider['credentials']['username'].replace('\\', '\\\\'),
+            provider['credentials']['password']))
+        ssh_client.run_command("cd /var/www/miq/vmdb; echo '" + command + "'"
+            "| bundle exec bin/rails c 2> /dev/null")
```
Comment (akrzos) [inline cfme-performance/utils/providers.py]: Matt I modified it for readability and found I could add the provider without the sleep statement through I am still concerned of a race condition in "slower" environments, but I'm not entire sure that is an issue yet.

``` python
    # Workaround for Bug: https://bugzilla.redhat.com/show_bug.cgi?id=1351253
    if provider['type'] == "ManageIQ::Providers::Microsoft::InfraManager":
        logger.debug('Running rails command for Microsoft InfraManager Workaround')
        ssh_client = SSHClient()
        command = (
            'm = ExtManagementSystem.find_by_name \'{}\';'
            'attributes = {{:security_protocol => \'{}\'}};'
            'm.update_authentication(:default => {{:userid => \'{}\', :password => \'{}\'}});'
            'm.update_attributes(attributes);'
            'm.save;'
            'm.authentication_check;'.format(provider['name'], provider['security_protocol'],
                provider['credentials']['username'].replace('\\', '\\\\'),
                provider['credentials']['password']))
        ssh_client.run_rails_console(command, timeout=None, log_less=True)
```

**Pull request merged**